lib/progress.ts

```diff
@@ -77,7 +77,7 @@ export interface IPublishProgress {
   /**
    * How far along are we?
    */
-  readonly percentComplete: number;
+  readonly percentComplete?: number;
```

Comment:
why?

Reply:
Percent complete is just a measure of how many assets and it's not relevant to what the shell logs are doing so we ignore it when capturing shell logs

Reply:
This strongly indicates to me that you are not using the interface called `IPublishProgress` correctly which defines itself as "Context object for publishing progress".